static/app/views/insights/common/components/issues.tsx

```diff
@@ -0,0 +1,264 @@
+import GroupChart from 'sentry/components/stream/groupChart';
+import {IconUser} from 'sentry/icons';
+import {t, tct, tn} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Group} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {IssueSummary} from 'sentry/views/performance/newTraceDetails/traceDrawer/details/issues/issueSummary';
+
+const TABLE_WIDTH_BREAKPOINTS = {
+  FIRST: 800,
+  SECOND: 600,
+  THIRD: 500,
+  FOURTH: 400,
+};
+
+function Issue({data}: {data: Group}) {
+  const organization = useOrganization();
+
+  return (
+    <StyledPanelItem as="tr">
+      <IssueSummaryWrapper>
+        <IssueSummary data={data} organization={organization} event_id={'0'} />
+        <EventOrGroupExtraDetails data={data} />
+      </IssueSummaryWrapper>
+      <ChartWrapper>
+        <GroupChart
+          stats={data.filtered ? data.filtered.stats?.['24h'] : data.stats?.['24h']}
+          secondaryStats={data.filtered ? data.stats?.['24h'] : []}
+          showSecondaryPoints
+          showMarkLine
+        />
+      </ChartWrapper>
+      <EventsWrapper>
+        <PrimaryCount value={data.filtered ? data.filtered.count : data.count} />
+      </EventsWrapper>
+      <UserCountWrapper>
+        <PrimaryCount value={data.filtered ? data.filtered.userCount : data.userCount} />
+      </UserCountWrapper>
+      <AssineeWrapper>
+        {data.assignedTo ? (
+          <ActorAvatar actor={data.assignedTo} hasTooltip size={24} />
+        ) : (
+          <StyledIconWrapper>
+            <IconUser size="md" />
+          </StyledIconWrapper>
+        )}
+      </AssineeWrapper>
+    </StyledPanelItem>
+  );
+}
+
+function IssueListHeader({issues}: {issues?: Group[]}) {
+  return (
+    <StyledPanelHeader as="tr">
+      <IssueHeading>
+        {tct(`[count] [text]`, {
+          count: issues?.length ?? 0,
+          text: tn('Related Issue', 'Related Issues', issues?.length ?? 0),
+        })}
+      </IssueHeading>
+
+      <GraphHeading>{t('Graph')}</GraphHeading>
+      <EventsHeading>{t('Events')}</EventsHeading>
+      <UsersHeading>{t('Users')}</UsersHeading>
+      <AssigneeHeading>{t('Assignee')}</AssigneeHeading>
+    </StyledPanelHeader>
+  );
+}
+
+function fetchIssues(
+  issueTypes: string[],
+  message?: string
+): {isLoading: boolean; issues?: Group[]} {
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+
+  let query = `issue.type:[${issueTypes.join(',')}]`;
+  // note: backend supports a maximum number of characters for message (seems to vary).
+  // so, we query the first 200 characters of `message`, then filter for exact `message`
+  // matches in application code
+  if (message) {
+    query += ` message:"${message?.slice(0, 200).replaceAll('"', '\\"')}"`;
+  }
```

Comment:
I'm not sure that we should proceed when `message` is falsey. I don't think it makes sense to make a query without filtering on the message, because it will just return a random performance issue. I'd suggest returning early when `message` is falsey instead.

Reply:
Couldn't return early here because the useApiQuery hook has to run at the top-level, but added a return and disabled the api query when message is falsey.